interceptors/payable_resource_authentication.go

```diff
@@ -40,7 +41,7 @@ func (payableAuthInterceptor *PayableAuthenticationInterceptor) PayableAuthentic
 
 		// Get identity type from request
 		identityType := authentication.GetAuthorisedIdentityType(r)
-		if !(identityType == authentication.Oauth2IdentityType || identityType == authentication.APIKeyIdentityType) {
+		if isUnauthorizedIdentityType(identityType) {
```

Comment:
Attempt to reduce code complexity by group multiple logic checks into one go function